src/components/Track/index.ts

```diff
@@ -0,0 +1,20 @@
+import { ReactElement, cloneElement } from 'react'
+
+const TRACK_ID_PROP = 'data-track-id'
+const TRACK_PAYLOAD_PROP = 'data-track-payload'
+
+type Props = {
+  children: ReactElement
+  [TRACK_ID_PROP]: string
+  [TRACK_PAYLOAD_PROP]: unknown
+}
+
+const Track = ({ children, [TRACK_ID_PROP]: trackId, [TRACK_PAYLOAD_PROP]: trackPayload }: Props): ReactElement => {
+  return cloneElement(children, {
+    ...children.props,
+    [TRACK_ID_PROP]: trackId,
+    [TRACK_PAYLOAD_PROP]: trackPayload ? JSON.stringify(trackPayload) : undefined,
```

Comment:
Please make sure `data-track-payload="undefined"` doesn't end up in the resulting DOM.